src/enforcer/HolographERC20.sol

```diff
@@ -341,17 +341,22 @@ contract HolographERC20 is Admin, Owner, Initializable, NonReentrant, EIP712, Ho
         amount
       );
       assembly {
-        mstore(add(sourcePayload, mload(sourcePayload)), caller())
+        // it is important to add 32 bytes in order to accommodate the first 32 bytes being used for storing length of bytes
+        mstore(add(sourcePayload, add(mload(sourcePayload), 0x20)), caller())
         let result := call(
           gas(),
           sload(_sourceContractSlot),
           callvalue(),
-          sourcePayload,
-          add(mload(sourcePayload), 32),
```

Comment:
Just wondering if the original intentions here were to skip 32 bytes but it needs to be done in hex via 0x20 instead?

Reply:
The use of 32 here was correct. Switched to 0x20 just for clarity and to not confuse bytes with integers.